src/app/events/[slug]/utils/filterAndSortScheduleDays.ts

```diff
@@ -0,0 +1,28 @@
+import { UTCDate } from '@date-fns/utc'
+
+import type { Event } from '../../types/eventType'
+
+export function filterAndSortScheduleDays(
+  schedule: NonNullable<Event['schedule']>,
+) {
+  {
+    const daysWithEvents = schedule.days.filter((day) => day.events.length > 0)
+
+    const daysWithEventsSortedByTime = daysWithEvents.map((day) => ({
+      ...day,
+      events: day.events.sort((a, b) => {
+        const startA = new UTCDate(a.start).getTime()
+        const startB = new UTCDate(b.start).getTime()
+        return startA - startB
+      }),
+    }))
+
+    const sortedDays = daysWithEventsSortedByTime.sort((a, b) => {
```

Comment:
Nit: Best to create a new array with `.sort` as it's doesn't do it for us, unlike most Array methods.

```suggestion
    const sortedDays = [...daysWithEventsSortedByTime].sort((a, b) => {
```

Reply:
TIL there's a new [toSorted()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted) method available in JavaScript, which is the same as `sort()` but doesn't update the original array!

No need to change anything, just sharing :)

Reply:
Oh - Amazing! Let's use it!

Reply:
Done! Thank you, @CharlyMartin!